README.md

```diff
@@ -1 +1,296 @@
+   - Replace `:categoryId` with the actual category ID.
+
+4. **Add a new product (Admin only):**
+   - **Endpoint:** POST `/products`
+   - **Body:**
+     ```json
+     {
+       "categoryId": "Category_ID",
+       "name": "Product Name",
+       "price": 100.50
+     }
+     ```
+
+5. **Update a product (Admin only):**
+   - **Endpoint:** PUT `/products/:id`
+   - Replace `:id` with the actual product ID you want to update.
+   - **Body:**
+     ```json
+     {
+       "categoryId": "New_Category_ID",
+       "name": "Updated Product Name",
+       "price": 105.00
+     }
+     ```
+
+6. **Delete a product (Admin only):**
+   - **Endpoint:** DELETE `/products/:id`
+   - Replace `:id` with the actual product ID you want to delete.
+
+### Categories
+
+1. **Get all categories:**
+   - **Endpoint:** GET `/categories`
+
+2. **Add a new category (Admin only):**
+   - **Endpoint:** POST `/categories`
+   - **Body:**
+     ```json
+     {
+       "name": "Category Name"
+     }
+     ```
+
+3. **Update a category (Admin only):**
+   - **Endpoint:** PUT `/categories/:id`
+   - Replace `:id` with the actual category ID you want to update.
+   - **Body:**
+     ```json
+     {
+       "name": "Updated Category Name"
+     }
+     ```
+
+4. **Delete a category (Admin only):**
+   - **Endpoint:** DELETE `/categories/:id`
+   - Replace `:id` with the actual category ID you want to delete.
+
+### Authorization
+
+Remember to include the JWT token in the request headers for routes that require authentication:
+```bash
+Authorization: Bearer YOUR_JWT_TOKEN_HERE
+```
+
+### Role-Based Access Control
+
+Only users with the `admin` role can create, update, and delete products and categories. Both `admin` and `customer` roles can view products and categories.
+
+
+## GraphQL Interface
+
+If you've integrated GraphQL, you can access the GraphQL playground at:
+
+```
+http://localhost:3000/graphql
+```
+
+
+## GraphQL Authorization and Usage
+
+### Authorizing Your GraphQL Requests
+
+After obtaining the JWT token from the REST API (by registering a new user or logging in), you need to add it to your GraphQL requests to access protected routes.
+
+1. **Obtain JWT Token**: Utilize the REST API to obtain the JWT token by registering or logging in via Postman using the provided collection.
+
+2. **Add to GraphQL Playground**:
+    - Navigate to [http://localhost:3000/graphql](http://localhost:3000/graphql).
+    - On the bottom left, find the "HTTP HEADERS" tab and insert your token like so:
+      ```json
+      {
+        "Authorization": "Bearer YOUR_JWT_TOKEN_HERE"
+      }
+      ```
+   Replace `YOUR_JWT_TOKEN_HERE` with the token you obtained.
+
+
+### Sample Queries and Mutations
+
+#### Queries
+
+1. **Fetch All Categories**
+   ```graphql
+   query {
+     categories {
+       id
+       name
+     }
+   }
+   ```
+
+2. **Fetch Specific Category**
+   ```graphql
+   query {
+     category(id: 1) {
+       id
+       name
+     }
+   }
+   ```
```

Comment:
Doesn't look like it's implemented!? (probably because it's not mentioned in the chapter)